gems/sorbet-runtime/lib/types/types/fixed_hash.rb

```diff
@@ -16,6 +16,21 @@ def name
       "{#{@types.map {|(k, v)| "#{k}: #{v}"}.join(', ')}}"
     end
 
+    # @override Base
+    def recursively_valid?(obj)
+      return false unless obj.is_a?(Hash)
+
+      @types.each do |key, type|
+        return false unless type.recursively_valid?(obj[key])
+      end
+
+      obj.each_key do |key|
+        return false unless @types[key]
+      end
+
+      true
+    end
```

Comment:
Feel free to push back here. Do you think this method could be improved by implementing it something like

```ruby
valid?(obj) && <logic for recursively_valid>
```

?

(motivation: looks like a lot of duplication with `valid?`)

Reply:
I was going to push back for performance concerns (since it'll re-traverse the hash twice, once for `valid?` and once for `recursively_valid?`) but on reflection, if you're using `recursively_valid?`, then you're already throwing perf out the window, so I'll do this refactor.